packages/text-area/test/text-area.test.js

```diff
@@ -0,0 +1,372 @@
+import { expect } from '@esm-bundle/chai';
+import sinon from 'sinon';
+import { fixtureSync, oneEvent } from '@vaadin/testing-helpers';
+import { TextArea } from '../src/vaadin-text-area.js';
+
+customElements.define('vaadin-text-area', TextArea);
+
+describe('text-area', () => {
+  let textArea;
+
+  beforeEach(() => {
+    textArea = fixtureSync('<vaadin-text-area></vaadin-text-area>');
+  });
+
+  describe('properties', () => {
+    let native;
+
+    beforeEach(() => {
+      native = textArea._inputNode;
+    });
+
+    describe('native', () => {
+      function assertAttrCanBeSet(prop, value) {
+        textArea[prop] = value;
+        const attrValue = native.getAttribute(prop);
+
+        if (value === true) {
+          expect(attrValue).not.to.be.null;
+        } else if (value === false) {
+          expect(attrValue).to.be.null;
+        } else if (value) {
+          expect(attrValue).to.be.equal(String(value));
+        }
+      }
+
+      function assertPropCanBeSet(prop, value) {
+        for (let i = 0; i < 3; i++) {
+          // Check different values
+          const newValue = typeof value === 'boolean' ? i % 2 === 0 : value + i;
+          textArea[prop] = newValue;
+          expect(native[prop]).to.be.equal(newValue);
+        }
+      }
+
+      ['placeholder', 'value'].forEach((prop) => {
+        it('should set string property ' + prop, () => {
+          assertPropCanBeSet(prop, 'foo');
+        });
+      });
+
+      ['disabled'].forEach((prop) => {
+        it('should set boolean property ' + prop, () => {
+          assertPropCanBeSet(prop, true);
+        });
+      });
+
+      ['maxlength', 'minlength'].forEach((prop) => {
+        it('should set numeric attribute ' + prop, () => {
+          assertAttrCanBeSet(prop, 2);
+        });
+      });
+
+      ['autocomplete'].forEach((prop) => {
+        it('should set boolean attribute ' + prop, () => {
+          assertAttrCanBeSet(prop, 'on');
+        });
+      });
+
+      ['autocapitalize'].forEach((prop) => {
+        it('should set boolean attribute ' + prop, () => {
+          assertAttrCanBeSet(prop, 'none');
+        });
+      });
+
+      ['autocomplete', 'autocorrect', 'readonly', 'required'].forEach((prop) => {
+        it('should set boolean attribute ' + prop, () => {
+          assertAttrCanBeSet(prop, true);
+          assertAttrCanBeSet(prop, false);
+        });
+      });
+
+      describe('input field', () => {
+        let inputField;
+
+        beforeEach(() => {
+          inputField = textArea.shadowRoot.querySelector('[part="input-field"]');
+        });
+
+        it('should propagate invalid property to the input container', () => {
```

Comment:
I don't understand why this is useful, can you explain why we want this behaviour?

Reply:
There is a test in the input container to ensure this property is reflected to attribute for styling:
https://github.com/vaadin/web-components/blob/b955ae51adbe9424bc604496c71093aad25923fe/packages/input-container/test/input-container.test.js#L32